projects/github-actions/repo-gardening/src/tasks/gather-support-references/index.js

```diff
@@ -0,0 +1,165 @@
+ * @param {string} owner   - Repository owner.
+ * @param {string} repo    - Repository name.
+ * @param {string} number  - Issue number.
+ * @returns {Promise<number>} Promise resolving to a number.
+ */
+async function getListComment( octokit, owner, repo, number ) {
+	let commentID = 0;
+
+	debug( `gather-support-references: Looking for a previous comment from this task in our issue.` );
+
+	for await ( const response of octokit.paginate.iterator( octokit.rest.issues.listComments, {
+		owner,
+		repo,
+		issue_number: +number,
+	} ) ) {
+		response.data.map( comment => {
+			if (
+				comment.user.login === 'github-actions[bot]' &&
+				comment.body.includes( '**Support References**' )
+			) {
+				commentID = comment.id;
+			}
+		} );
+	}
+
+	return commentID;
+}
+
+/**
+ * Scan the contents of the issue as well as all its comments, get all support references, and add them to an array of references.
+ *
+ * @param {GitHub} octokit - Initialized Octokit REST client.
+ * @param {string} owner   - Repository owner.
+ * @param {string} repo    - Repository name.
+ * @param {string} number  - Issue number.
+ * @returns {Promise<Array>} Promise resolving to an array.
+ */
+async function getIssueReferences( octokit, owner, repo, number ) {
+	const ticketReferences = [];
+	const supportIds = [];
+	const referencesRegexP = /[0-9]*-(?:chat|hc|zen|zd)/gim;
+
+	debug( `gather-support-references: Getting references from comments.` );
+	for await ( const response of octokit.paginate.iterator( octokit.rest.issues.listComments, {
+		owner: owner.login,
+		repo,
+		issue_number: +number,
+	} ) ) {
+		response.data.map( comment => {
+			ticketReferences.push( ...comment.body.matchAll( referencesRegexP ) );
+		} );
+	}
+
+	debug( `gather-support-references: Getting references from issue body.` );
+	const {
+		data: { body },
+	} = await octokit.rest.issues.get( {
+		owner: owner.login,
+		repo,
+		issue_number: +number,
+	} );
+	ticketReferences.push( ...body.matchAll( referencesRegexP ) );
+
+	// Buid a first array with only the support IDs we've collected.
+	ticketReferences.map( reference => {
+		supportIds.push( reference[ 0 ] );
+	} );
+
+	// That array can still include duplicates, or references that are not formatted quite properly.
+	// Let's build a final array with unique and correct support IDs.
+	const correctedSupportIds = [];
+	supportIds.map( supportId => {
+		// xxx-zen and xxx-hc are the preferred formats for tickets and chats.
+		// xxx-zd and xxx-chat, as well as uppercase versions, are considered as alternate versions.
+		const wrongId = supportId.match( /([0-9]*)-(zd|chat)/i );
+		if ( wrongId ) {
+			const correctedId = `${ wrongId[ 1 ] }-${ wrongId[ 2 ] === 'zd' ? 'zen' : 'hc' }`;
+			if ( ! correctedSupportIds.includes( correctedId ) ) {
+				correctedSupportIds.push( correctedId );
+			}
+		} else if ( ! correctedSupportIds.includes( supportId.toLowerCase() ) ) {
+			correctedSupportIds.push( supportId.toLowerCase() );
+		}
+	} );
```

Comment:
You could do
```
const correctedSupportIds = new Set();
```
then you can just `correctedSupportIds.add()` instead of having to guard it with `if ( ! correctedSupportIds.includes( correctedId ) )`.

Converting back to an array can be done with `[ ...correctedSupportIds ]` if you need to.

You might also switch from `.map()` to `.forEach()`, since you're not using the return value. Or you could so far as to do `for ( const supportId of supportIds )` instead, but I suspect readability of that versus `.forEach()` is a matter of taste.

Reply:
It does simplify things a bit. 👍 done in 051ed1783e30eeb1363247b3c4f0a71c84893e72